### ISSUE TITLE
Problem with remote transmit

### ISSUE THREAD

**Issue by ab316** [101100/xbee-rx#9]
I am running the simple-transmit sample to test transmission of my Xbee S2 module.
However, I am not getting the "Transmission Successful" response in the console.
Playing with the xbee-rx.js file, I found that in the `_remoteTransmit` function, there is this block of code:

```
return _sendFrameStreamResponse(frame, timeoutMs, responseFrameType)
	.flatMap(function (frame) {
		if (frame.deliveryStatus === xbee_api.constants.DELIVERY_STATUS.SUCCESS) {
			return rx.Observable.empty();
		}

		// if not OK, throw error
		return rx.Observable.throw(new Error(xbee_api.constants.DELIVERY_STATUS[frame.deliveryStatus]));
	});
```

If I change `flatMap` to `map` or If I return anything else instead of the empty Observable, then I start getting the "Transmission Successful" message.

What's the issue here? I haven't worked with RX before so I don't have any idea about it.

**Comment by 101100** [comment]
Ah, at some point I switched from generating `true` when the packet was successfully sent to just completing the stream (with no value in the stream).  Unfortunately, the example was never updated to match that behaviour.  If you change the last part of the example to the following you will get the correct result.

```js
xbee
    .remoteTransmit({
        destinationId: destinationId,
        data: data
    })
    .subscribe(function () {
        // a successful transmission generates no values
    }, function (e) {
        console.log("Transmission failed:\n", e);
        xbee.close();
    }, function () {
        console.log("Transmission successful");
        xbee.close();
    });
```

I think I'll update the library to return `true` again in version `3.0` since that will make it easier to combine multiple transmissions.

**Comment by ab316** [comment]
Okay got it!
Just one more question though. If the first argument function is called on a successful transmission then what's the third one for? Since it is also logging "Transmission Successful"

**Comment by 101100** [comment]
Rx streams can emit a sequence of values and then end with either an error or a completion.  The three functions are handling values, errors and then completions in that order.  So the first function does nothing, the second handles errors and the third handles completion, which in this case means success.

**Comment by 101100** [comment]
I just published version `3.0.1` which results in `true` for a successful transmission.  That should work with the original example as is.

**Comment by ab316** [comment]
Ok, great!
